src/de/grobox/liberario/fragments/ProductDialogFragment.java

```diff
@@ -108,7 +113,20 @@ public void onClick(View v) {
 				// call listener if set
 				if(listener != null) {
 					EnumSet<Product> products = getProductsFromItems(adapter.getSelectedItems());
-					listener.onProductsChanged(products);
+                    // get the current network
+                    TransportNetwork current_network = Preferences.getTransportNetwork(getContext());
+                    String network_pref = current_network + "_selected_products";
+                    SharedPreferences sp = getContext().getSharedPreferences("pref_selected_products", 0);
```

Comment:
No, if we do one shared preferences file per network that holds network specific settings, you could use something like `current_network.getId().toString()` for the name.